src/Microsoft.ComponentDetection.Detectors/dotnet/DotNetComponentDetector.cs

```diff
@@ -58,8 +58,22 @@ public DotNetComponentDetector(
 
     public override IEnumerable<string> Categories => ["DotNet"];
 
+    private static string TrimAllEndingDirectorySeparators(string path)
+    {
+        string last;
+
+        do
+        {
+            last = path;
+            path = Path.TrimEndingDirectorySeparator(last);
+        }
+        while (!ReferenceEquals(last, path));
+
+        return path;
+    }
```

Comment:
I wonder if it'd be easier to just do `path.TrimEnd(Path.DirectorySeparatorChar).TrimEnd(Path.AltDirectorySeparatorChar)`